apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/StatementInstrumentation.java

```diff
@@ -76,7 +76,8 @@
 
     @Override
     public ElementMatcher<? super NamedElement> getTypeMatcherPreFilter() {
-        return nameContains("Statement");
+        // DB2 driver does not call its Statements statements.
+        return nameContains("Statement").or(nameStartsWith("com.ibm.db2.jcc"));
```

Comment:
Maybe just checking if the name contains `Connection` could be more generic? Both is fine to me, I leave up to you 🙂 

```suggestion
        return nameContains("Statement").or(nameContains("Connection"));
```

Reply:
The user reported what we missed is called `com.ibm.db2.jcc.am.k4`